Zend/zend_API.c

```diff
@@ -1193,7 +1210,19 @@ ZEND_API void object_properties_init_ex(zend_object *object, HashTable *properti
 			    property_info &&
 			    (property_info->flags & ZEND_ACC_STATIC) == 0) {
 				zval *slot = OBJ_PROP(object, property_info->offset);
-				ZVAL_COPY_VALUE(slot, prop);
+
+				if (UNEXPECTED(property_info->type)) {
+					zval tmp, *val;
+
+					val = zend_verify_property_type(property_info, prop, &tmp, 0);
+					if (UNEXPECTED(!val)) {
+						zend_verify_property_type_error(property_info, key, prop);
+						continue;
```

Comment:
I don't know. internal functions may clear the exception. So, probably yes. And it doesn't do any harm either.